dell/src/test/java/org/apache/iceberg/dell/mock/ecs/EcsS3MockRule.java

```diff
@@ -33,16 +33,16 @@
 import org.apache.iceberg.dell.DellClientFactories;
 import org.apache.iceberg.dell.DellProperties;
 import org.apache.iceberg.dell.mock.MockDellClientFactory;
-import org.junit.rules.TestRule;
-import org.junit.runner.Description;
-import org.junit.runners.model.Statement;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 
 /**
  * Mock rule of ECS S3 mock.
```

Comment:
nit: should we update the javadoc too?

Reply:
+1 on updating javadoc

Reply:
I am thinking of adding these annotations in [this section ?](https://github.com/apache/iceberg/blob/master/CONTRIBUTING.md#junit4--junit5) Does that help or there's some other place I need to update in?

Reply:
as part of this PR I think what makes sense is to change the javadoc here from `Mock rule of ECS S3 mock` to `Test extension for using an ECS S3 mock`

Reply:
We can also update the class name from ```EcsS3MockRule``` to ```EcsS3MockExtension``` at all the places for better readability.